[Bug]: RP2040 - Wire.begin does not accept two arguments with SDA/SCL pins

### Category

Hardware Compatibility

### Hardware

Raspberry Pi Pico (W)

### Firmware Version

2.2.17

### Description

The actual firmware doesn't build for RP2040 when the I2C_0 or I2C_1 pins are redefined via I2C_SDA or I2C_SDA1.

In main.cpp, line 372:
```C
#ifdef I2C_SDA1
    Wire1.begin(I2C_SDA1, I2C_SCL1);
#endif

#ifdef I2C_SDA
    Wire.begin(I2C_SDA, I2C_SCL);
#elif HAS_WIRE
    Wire.begin();
#endif
```
and line 426:
```C
#ifdef I2C_SDA1
    Wire1.begin(I2C_SDA1, I2C_SCL1);
    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE1);
#endif

#ifdef I2C_SDA
    Wire.begin(I2C_SDA, I2C_SCL);
    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE);
#elif HAS_WIRE
    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE);
#endif
```
the _Wire.begin_ function is called with two arguments to set the SDA and SCL pins. However, the Arduino Pico Framework library Wire has no _begin_ that accept two arguments for pin change. Done via _setSDA_ and _setSCL_.

Not sure if other hardware needs that specific _Wire.begin_ call but a possible fix for the RP2040 might be:
```C
#if defined(I2C_SDA1) && defined(ARCH_RP2040)
    Wire1.setSDA(I2C_SDA1);
    Wire1.setSCL(I2C_SCL1);
    Wire1.begin();
#elif defined(I2C_SDA1) && !defined(ARCH_RP2040)
    Wire1.begin(I2C_SDA1, I2C_SCL);
#endif

#if defined(I2C_SDA) && defined(ARCH_RP2040)
    Wire.setSDA(I2C_SDA);
    Wire.setSCL(I2C_SCL);
    Wire.begin();
#elif defined(I2C_SDA) && !defined(ARCH_RP2040)
    Wire.begin(I2C_SDA, I2C_SCL);
#elif HAS_WIRE
    Wire.begin();
#endif
```
```C
#if defined(I2C_SDA1) && defined(ARCH_RP2040)
    Wire1.setSDA(I2C_SDA1);
    Wire1.setSCL(I2C_SCL1);
    Wire1.begin();
    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE1);
#elif defined(I2C_SDA1) && !defined(ARCH_RP2040)
    Wire1.begin(I2C_SDA1, I2C_SCL1);
    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE1);
#endif

#if defined(I2C_SDA) && defined(ARCH_RP2040)
    Wire.setSDA(I2C_SDA);
    Wire.setSCL(I2C_SCL);
    Wire.begin();
    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE);
#elif defined(I2C_SDA) && !defined(ARCH_RP2040)
    Wire.begin(I2C_SDA, I2C_SCL);
    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE);
#elif HAS_WIRE
    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE);
#endif
```

### Relevant log output

```Shell
Using meshtastic platformio-custom.py, firmware version 2.2.17.db8f8db8
Compiling .pio/build/pico_waveshare/src/main.cpp.o
src/main.cpp: In function 'void setup()':
src/main.cpp:373:16: error: no matching function for call to 'TwoWire::begin(int, int)'
  373 |     Wire1.begin(I2C_SDA1, I2C_SCL1);
      |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In file included from .pio/libdeps/pico_waveshare/Adafruit BusIO/Adafruit_I2CDevice.h:5,
                 from .pio/libdeps/pico_waveshare/Adafruit BusIO/Adafruit_BusIO_Register.h:9,
                 from .pio/libdeps/pico_waveshare/Adafruit INA219/Adafruit_INA219.h:21,
                 from src/modules/Telemetry/Sensor/INA219Sensor.h:4,
                 from src/power.h:26,
                 from src/main.cpp:11:
/home/lupus/.platformio/packages/framework-arduinopico/libraries/Wire/src/Wire.h:42:10: note: candidate: 'virtual void TwoWire::begin()'
   42 |     void begin() override;
      |          ^~~~~
/home/lupus/.platformio/packages/framework-arduinopico/libraries/Wire/src/Wire.h:42:10: note:   candidate expects 0 arguments, 2 provided
/home/lupus/.platformio/packages/framework-arduinopico/libraries/Wire/src/Wire.h:44:10: note: candidate: 'virtual void TwoWire::begin(uint8_t)'
   44 |     void begin(uint8_t address) override;
      |          ^~~~~
/home/lupus/.platformio/packages/framework-arduinopico/libraries/Wire/src/Wire.h:44:10: note:   candidate expects 1 argument, 2 provided
src/main.cpp:427:16: error: no matching function for call to 'TwoWire::begin(int, int)'
  427 |     Wire1.begin(I2C_SDA1, I2C_SCL1);
      |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
/home/lupus/.platformio/packages/framework-arduinopico/libraries/Wire/src/Wire.h:42:10: note: candidate: 'virtual void TwoWire::begin()'
   42 |     void begin() override;
      |          ^~~~~
/home/lupus/.platformio/packages/framework-arduinopico/libraries/Wire/src/Wire.h:42:10: note:   candidate expects 0 arguments, 2 provided
/home/lupus/.platformio/packages/framework-arduinopico/libraries/Wire/src/Wire.h:44:10: note: candidate: 'virtual void TwoWire::begin(uint8_t)'
   44 |     void begin(uint8_t address) override;
      |          ^~~~~
/home/lupus/.platformio/packages/framework-arduinopico/libraries/Wire/src/Wire.h:44:10: note:   candidate expects 1 argument, 2 provided
*** [.pio/build/pico_waveshare/src/main.cpp.o] Error 1
```
```

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -369,11 +369,19 @@ void setup()
 
 #endif
 
-#ifdef I2C_SDA1
-    Wire1.begin(I2C_SDA1, I2C_SCL1);
+#if defined(I2C_SDA1) && defined(ARCH_RP2040)
+    Wire1.setSDA(I2C_SDA1);
+    Wire1.setSCL(I2C_SCL1);
+    Wire1.begin();
+#elif defined(I2C_SDA1) && !defined(ARCH_RP2040)
+    Wire1.begin(I2C_SDA1, I2C_SCL);
 #endif
 
-#ifdef I2C_SDA
+#if defined(I2C_SDA) && defined(ARCH_RP2040)
+    Wire.setSDA(I2C_SDA);
+    Wire.setSCL(I2C_SCL);
+    Wire.begin();
+#elif defined(I2C_SDA) && !defined(ARCH_RP2040)
     Wire.begin(I2C_SDA, I2C_SCL);
 #elif HAS_WIRE
     Wire.begin();
@@ -423,12 +431,22 @@ void setup()
 
     LOG_INFO("Scanning for i2c devices...\n");
 
-#ifdef I2C_SDA1
+#if defined(I2C_SDA1) && defined(ARCH_RP2040)
+    Wire1.setSDA(I2C_SDA1);
+    Wire1.setSCL(I2C_SCL1);
+    Wire1.begin();
+    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE1);
+#elif defined(I2C_SDA1) && !defined(ARCH_RP2040)
     Wire1.begin(I2C_SDA1, I2C_SCL1);
     i2cScanner->scanPort(ScanI2C::I2CPort::WIRE1);
 #endif
 
-#ifdef I2C_SDA
+#if defined(I2C_SDA) && defined(ARCH_RP2040)
+    Wire.setSDA(I2C_SDA);
+    Wire.setSCL(I2C_SCL);
+    Wire.begin();
+    i2cScanner->scanPort(ScanI2C::I2CPort::WIRE);
+#elif defined(I2C_SDA) && !defined(ARCH_RP2040)
     Wire.begin(I2C_SDA, I2C_SCL);
     i2cScanner->scanPort(ScanI2C::I2CPort::WIRE);
 #elif HAS_WIRE

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -374,7 +374,7 @@ void setup()
     Wire1.setSCL(I2C_SCL1);
     Wire1.begin();
 #elif defined(I2C_SDA1) && !defined(ARCH_RP2040)
-    Wire1.begin(I2C_SDA1, I2C_SCL);
+    Wire1.begin(I2C_SDA1, I2C_SCL1);
 #endif
 
 #if defined(I2C_SDA) && defined(ARCH_RP2040)

